js/client-battle-tooltips.js

```diff
@@ -37,11 +37,71 @@ var BattleTooltips = (function () {
-	};
+	if (navigator.userAgent.match(/(Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini)/)) {
+		// If user is using a mobile phone or tablet, use touch events
+
+		// Touch delay, pressing finger more than that time will cause the tooltip to open.
+		// Shorter time will cause teh button to click
+		var touchDelayForTap = 300; // ms
+
+		// Each time a finger presses the button this function will be callled
+		// First finger starts the counter, and when last fingers leaves time is checked
+		BattleTooltips._onTouchStart = function (roomid, thing, type, elem, ownHeight) {
+			// Saving number of Touch start happends (number of fingers on teh button,
+		  // though most of the time will be 1, don't won't a nasty bug like tooltip stuck open)
+			// Starts by 1 and increases each time this func is called
+			elem._numOfTaps = elem._numOfTaps !== undefined ? (elem._numOfTaps + 1) : 1;
+
+			// On first tap start counting
+			if (elem._numOfTaps === 1) {
+				setTimeout(function () {
+					// When tap is undefined means `_onTouchEnd` deleted it by the end of the time, perform the action
+					if (elem._numOfTaps === undefined) {
+						// short tap
+						elem._isCalledFromTocuh = true;
+						elem.click();
+					} else {
+						// When tap is not defined open the tooltip until `_onTouchEnd` will close it
+						elem._toolTipIsOpen = true;
+						BattleTooltips.showTooltipFor(roomid, thing, type, elem, ownHeight);
+					}
+				}, touchDelayForTap);
+			}
+		};
+		BattleTooltips._onTouchEnd = function (elem) {
+			// _numOfTaps always defined by `_onTouchStart` - is 1 or higher
+			elem._numOfTaps--;
+			// Delete variable when last finger left, no need for memory
+			if (elem._numOfTaps === 0) {
+				delete elem._numOfTaps;
+			}
 
+			// If tooltip is open and the last finger left, close the tooptip
+			if (elem._toolTipIsOpen && elem._numOfTaps === undefined) {
+				// Delete variable
+				delete elem._toolTipIsOpen;
+				BattleTooltips.hideTooltip();
+			}
+		};
+		// We define
+		BattleTooltips._onClick = function (elem, e) {
+			if (!elem._isCalledFromTocuh) {
+				e.stopPropagation();
+			} else {
+				delete elem._isCalledFromTocuh;
+			}
+		};
+		BattleTooltips.prototype.tooltipAttrs = function (thing, type, ownHeight) {
+			var roomid = this.room.id;
+			return ' onclick="BattleTooltips._onClick(this, event)" ontouchstart="BattleTooltips._onTouchStart(\'' + roomid + '\', \'' + Tools.escapeHTML('' + thing, true) + '\',\'' + type + '\', this, ' + (ownHeight ? 'true' : 'false') + ')" ontouchend="BattleTooltips._onTouchEnd(this)" ontouchleave="BattleTooltips._onTouchEnd(this)"';
+		};
+	} else {
+		// If user is using a desktop, use mouse events
```

Comment:
Yeah you are right, better support both (I read [this](http://www.html5rocks.com/en/mobile/touchandmouse/) too).

Might add a variable like I did with `_isCalledFromTocuh` that in `_onTouchStart` to `onMouseOver`, unless there is a better way to call that action.